DataFormats/PatCandidates/src/TriggerObjectStandAlone.cc

```diff
@@ -413,12 +413,10 @@ std::vector<std::string> const *TriggerObjectStandAlone::allLabels(edm::Paramete
     const unsigned int n(triggerNames.size());
     std::set<std::string> saveTags;
     for (unsigned int i = 0; i != n; ++i) {
-      if (pset->existsAs<vector<string> >(triggerNames.triggerName(i), true)) {
-        auto modules = pset->getParameter<vector<string> >(triggerNames.triggerName(i));
-        for (size_t m = 0; m < modules.size(); m++) {
-          auto module = modules[m];
-          auto moduleStrip = module.front() != '-' ? module : module.substr(1);
-
+      if (pset->existsAs<vector<string>>(triggerNames.triggerName(i), true)) {
+        auto const &modules = pset->getParameter<vector<string>>(triggerNames.triggerName(i));
+        for (auto const &module : modules) {
+          auto const moduleStrip = module.front() != '-' and module.front() != '!' ? module : module.substr(1);
```

Comment:
Naive comment: shouldn't this be a tilde, instead (or in addition)?

Reply:
So, what I did was simply print all these strings and it was clear that `cms.ignore(process.foo)` resolves to `-foo`, while `~process.foo` resolves to `!foo`.

I don't actually know where the creation of these strings happens: it would be useful to know, so that we can figure out if other special cases/characters should be kept into account for EDFilters.

A naive look at the code base with `git grep "'\!'"` did not give me certainty that `-` and `!` are the only two cases (maybe they are), e.g.
https://github.com/cms-sw/cmssw/blob/6abbda549d88a3999eaf90f941c1d93b997240cf/FWCore/ParameterSet/python/SequenceTypes.py#L576-L628
https://github.com/cms-sw/cmssw/blob/6abbda549d88a3999eaf90f941c1d93b997240cf/FWCore/Framework/src/StreamSchedule.cc#L397-L405